spec.html

```diff
@@ -9333,15 +9333,31 @@ <h1>Static Semantics: UTF16Encoding ( _cp_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-utf16decode" aoid="UTF16Decode">
-      <h1>Static Semantics: UTF16Decode ( _lead_, _trail_ )</h1>
+    <emu-clause id="sec-utf16encode" aoid="UTF16Encode">
+      <h1>Static Semantics: UTF16Encode ( _text_ )</h1>
+      <p>This abstract operation converts _text_, a sequence of Unicode code points, into a String value, as described in <emu-xref href="#sec-ecmascript-language-types-string-type"></emu-xref>.</p>
+      <emu-alg>
+        1. Return the string-concatenation of the code units that are the UTF16Encoding of each code point in _text_, in order.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-utf16decodesurrogatepair" aoid="UTF16DecodeSurrogatePair">
+      <h1>Static Semantics: UTF16DecodeSurrogatePair ( _lead_, _trail_ )</h1>
       <p>Two code units, _lead_ and _trail_, that form a UTF-16 <emu-xref href="#surrogate-pair"></emu-xref> are converted to a code point by performing the following steps:</p>
       <emu-alg>
         1. Assert: _lead_ is a <emu-xref href="#leading-surrogate"></emu-xref> and _trail_ is a <emu-xref href="#trailing-surrogate"></emu-xref>.
         1. Let _cp_ be (_lead_ - 0xD800) &times; 0x400 + (_trail_ - 0xDC00) + 0x10000.
         1. Return the code point _cp_.
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-utf16decode" aoid="UTF16Decode">
+      <h1>Static Semantics: UTF16Decode ( _string_ )</h1>
```

Comment:
I think this should be named UTF16DecodeString, both for clarity and to avoid colliding with the preexisting operation (which could therefore be decorated with `oldids="sec-utf16decode"`).

Reply:
That seems good for a number of reasons.

Reply:
I'd be okay with that. (The asymmetry with "UTF16Encode" offends my aesthetics slightly, but I can mash that down.)